### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by Omwomo**
- Removed current_user method in ApplicationController
- Installed and setup devise performing the following actions.
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- Modified the views in devise for registration and login to match the wireframes from the Sneak Peek

### CODE REVIEW

**Review by MahmoudBakr23 — COMMENTED** [review]
### Hi @Omwomo 😃, 

As much as I appreciate your hard work on this project, I'm afraid that I can't review this pull request.

**Invalid Submission**

- This is a pair programming activity, please make sure that all team members are going are committing to this PR.

_Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready._ 🙏

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

<hr>

_Invalid Code Review Request does not count into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,14 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (iambenkis) [inline app/models/user.rb]: - [ ] Based on the requirements of this project, this app should ask for email confirmations, hence `:confirmable` should be enabled in the user model and make sure that a confirmation message is sent to a new user after registering to confirm his/her email.

**Review by iambenkis — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Omwomo, @MahmoudBakr23   This is @iambenkis your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job trying harder, you are almost there

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@iambenkis] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/environments/development.rb [hunk 2]
```diff
@@ -41,6 +41,18 @@
...
+
+  config.action_mailer.delivery_method = :letter_opener
+
+  config.action_mailer.perform_deliveries = true
+
+  config.action_controller.raise_on_missing_callback_actions = true
+
+  LetterOpener.configure do |config|
+    config.browser = :chrome
+  end
```
Comment (franclobo) [inline config/environments/development.rb]: - [ ] It wasn't possible to open your app due to this issue, please 🙏🏽 correct this error to be able to check your app 👇🏽 

```ruby
bin/rails aborted!
NoMethodError: undefined method `browser=' for #<LetterOpener::Configuration:0x00007f54bcc002e8 @location=#<Pathname:/home/franclobo/microverse/eugene-mindscape/tmp/letter_opener>, @message_template="default", @file_uri_scheme=nil> (NoMethodError)

    config.browser = :chrome
          ^^^^^^^^^^
/home/franclobo/microverse/eugene-mindscape/config/environments/development.rb:53:in `block (2 levels) in <main>'
/home/franclobo/microverse/eugene-mindscape/config/environments/development.rb:52:in `block in <main>'
/home/franclobo/microverse/eugene-mindscape/config/environments/development.rb:3:in `<main>'
/home/franclobo/microverse/eugene-mindscape/config/environment.rb:5:in `<main>'
Tasks: TOP => db:drop => db:load_config => environment
(See full trace by running task with --trace)
```

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Omwomo, 

# Required Changes ♻️

istockphoto-1079018918-612x612

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️ 
- Use devise ✔️ 
- Descriptive PR ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Omwomo** [comment]
#### Incase the letter_opener gem is not functioning as expected on your machine, you can check the logs, which indicates the confirmations are being sent correctly and if you follow the link, you are being logged in successfully. 

Screenshot (1114)

**Review by Efiamotu-1 — APPROVED** [review]
Hi @Omwomo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**